sql/expression/comparison_test.go

```diff
@@ -210,6 +210,26 @@ func testRegexpCases(t *testing.T) {
 	}
 }
 
+func TestInvalidRegexp(t *testing.T) {
+	t.Helper()
+	require := require.New(t)
+
+	col1 := NewGetField(0, sql.Text, "col1", true)
+	require.NotNil(col1)
+
+	invalid := NewLiteral("*col1", sql.Text)
+	require.NotNil(invalid)
+
+	r := NewRegexp(col1, invalid)
+	require.NotNil(r)
```

Comment:
no need to check not nil, it will never be

Reply:
done